src/components/LegendPanel.vue

```diff
@@ -121,16 +123,25 @@ export default {
           return {};
         }
 
-        const namespace = layer.legend.namespace;
-        const styleName = layer.legend.style;
-        const layerId = layer.legend.layer;
-        const url = this.services && this.services.LEGEND_URL;
+        let legendImageSrc = '';
+
+        if (this.customMapConfig?.[layer.id]) {
+          legendImageSrc = this.customMapConfig[layer.id].legendImageSrc || '';
+        }
+
+        if (!legendImageSrc) {
+          const namespace = layer.legend.namespace;
+          const styleName = layer.legend.style;
+          const layerId = layer.legend.layer;
+          const url = this.services && this.services.LEGEND_URL;
+          legendImageSrc = url
+            ? `${url}/${namespace}/wms?REQUEST=GetLegendGraphic&VERSION=1.0.0&FORMAT=image/png&LAYER=${layerId}&STYLE=${styleName}&HEIGHT=16&WIDTH=16&LEGEND_OPTIONS=fontAntiAliasing:true;fontSize:14;mx:0;dx:10;fontName:Verdana;`
+            : '';
+        }
```

Comment:
If the customMapConfig isn't set correctly or there is anything wrong we still want to try and get a legend. It's should not occur, but it could go wrong. So thats why I just always try to get a legend if there isn't already a legend